src/components/ebay-pagination/index.js

```diff
@@ -107,12 +107,22 @@ function handlePageClick(event) {
 
 function handleNextPage(event) {
     eventUtils.preventDefaultIfHijax(event, this.state.hijax);
-    emitAndFire(this, 'pagination-next', { el: event.target });
+    emitAndFire(this, 'pagination-next', { el: getCorrectElement(event) });
 }
 
 function handlePreviousPage(event) {
     eventUtils.preventDefaultIfHijax(event, this.state.hijax);
-    emitAndFire(this, 'pagination-previous', { el: event.target });
+    emitAndFire(this, 'pagination-previous', { el: getCorrectElement(event) });
+}
+
+/**
+ * gets the correct target for pagination arrow button
+ */
+function getCorrectElement(event) {
```

Comment:
What about something like this, so it is a bit more re-useable? Let me know what you think.

```javascript
/**
 * @function getPreferredElement
 * @param {Event} event
 * @param {String} undesiredNode
 * @param {String} desiredNode
 */
function getPreferredElement(event, undesiredNode, desiredNode) {
  if(!(event instanceof Event)) {
    throw new Error(`Function ${arguments.callee.name}: Parameter 1 must be of type Event`);
  }

  if(typeof undesiredNode != 'string' || typeof desiredNode != 'string') {
    throw new Error(`Function ${arguments.callee.name}: Parameter 2 and 3 must be of type String`);
  }

  if (event.target.nodeName.toLowerCase() === desiredNode) {
    return event.target;
  }

  if (event.target.nodeName.toLowerCase() === undesiredNode) {
    const desiredElement = event.target.closest(desiredNode);

    if (desiredElement) {
      return desiredElement;
    }

    throw new Error(`Function ${arguments.callee.name}: Undesired element found but the desired element was not found`);
  }

  return event.target;
}

//example usage
const paginationLink = document.getElementById('#next-arrow');

paginationLink.addEventListener('submit', event => {
  const link = getPreferredElement(event, 'span', 'a');
  // ...
});
```

Reply:
Good one. Maybe we should add this to common utilities.

Reply:
A generic utility like this would be nice, but IMO this change is ok for a quick fix. Though, you would need to test it either way. We have similar logic in the menu:
https://github.com/eBay/ebayui-core/blob/5b5b22d37f89620b1280299a9834819f07205361/src/components/ebay-menu/index.js#L214-L217

https://github.com/eBay/ebayui-core/blob/5b5b22d37f89620b1280299a9834819f07205361/src/components/ebay-menu/test/test.browser.js#L270-L289

Reply:
I agree with Yoni. Let's keep this simple, as is, for now.